chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -217,16 +219,26 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
     }
 
     /// Read previous block from storage and return its BlockIndex.
-    fn get_previous_block_index(
+    fn get_previous_block_index_by_header(
         &self,
-        block_index: &BlockIndex,
+        block_header: &SignedBlockHeader,
     ) -> Result<GenBlockIndex, PropertyQueryError> {
-        let prev_block_id = block_index.prev_block_id();
+        let prev_block_id = block_header.prev_block_id();
         self.get_gen_block_index(prev_block_id)
             .log_err()?
             .ok_or(PropertyQueryError::PrevBlockIndexNotFound(*prev_block_id))
     }
 
+    /// Read previous block from storage and return its BlockIndex.
+    // FIXME: rename it? Or maybe delete it and rename get_previous_block_index_by_header to
+    // get_previous_block_index?
+    fn get_previous_block_index(
```

Comment:
I wouldn't care. When this PR is ready for review, let's see what others think of it.

Reply:
I'll leave it as it is for now

Reply:
Actually, after merging master into my branch I realized that I need a function that returns `CheckBlockError` if the parent wasn't found, because this error is now expected in certain tests. So, I reverted `get_previous_block_index` to the previous state and replaced `get_previous_block_index_by_header` with `get_previous_block_index_for_check_block`.